smarts/core/agent_interface.py

```diff
@@ -197,10 +197,20 @@ class AgentsAliveDoneCriteria:
 class ActorsAliveDoneCriteria:
     """Require actors to persist."""
 
-    actors_of_interest: Tuple[str, ...] = ()
+    actors_filter: Tuple[str, ...] = ()
     """Actors that should exist to continue this agent."""
 
     strict: bool = True
+    """If strict the agent will be done instantly if a target actor is not available
+    immediately.
+    """
```

Comment:
Should we remove `ActorsAliveDoneCriteria`, because we already have (i) `ScenarioInterestDoneCriteria` and (ii) `AgentsAliveDoneCriteria` ?

Reply:
They have slightly different uses. One is specified at the scenario generation level, the other solely in the interface. Maybe I could merge the two?

Reply:
Yes, do consider merging the underlying code for both of them.

Reply:
I merged the two. Now agent interface defined actors of interest also show up in the observations.